### PR TITLE
feat: endpoints for seedlot status

### PR DESCRIPTION
**Pull request by annibalsilva**
# Description

Create endpoints to fetch seed lot status, which will be all represented in the system by enums for now.

Fixes [FSADT2-491](https://apps.nrs.gov.bc.ca/int/jira/browse/FSADT2-491)

## Type of change

Please delete options that are not relevant.

- [x] New feature (non-breaking change which adds functionality)
- [x] This change requires a documentation update

# How Has This Been Tested?

- Manually
- Unit tests

## Checklist

- [x] I have read the [CONTRIBUTING](CONTRIBUTING.md) doc
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have already been accepted and merged

## Further comments

---
Thanks for the PR!
Any successful deployments (not always required) will be available below.
[Backend](https://nr-spar-backend-53-backend.apps.silver.devops.gov.bc.ca/) available
Once merged, code will be promoted and handed off to following workflow run.
[Main Merge Workflow](https://github.com/bcgov/nr-spar-backend/actions/workflows/merge-main.yml)

### CODE REVIEW

**Review by lucasbuenoencora — APPROVED** [review]
LGTM! Tested endpoints returns data successfully.

File: backend/src/main/java/ca/bc/gov/backendstartapi/endpoint/SeedlotStatusEndpoint.java [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+import ca.bc.gov.backendstartapi.enums.SeedlotStatus;
+import io.swagger.v3.oas.annotations.tags.Tag;
+import org.springframework.util.MimeTypeUtils;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+/** Endpoints to fetch {@link SeedlotStatus}. */
+@RestController
+@RequestMapping(path = "/api/seedlot-status", produces = MimeTypeUtils.APPLICATION_JSON_VALUE)
+@Tag(name = "SeedlotStatus")
```
Comment (RMCampos) [inline backend/src/main/java/ca/bc/gov/backendstartapi/endpoint/SeedlotStatusEndpoint.java:12]: I think here would be `SeedlotStatusEndpoint`

**Review by RMCampos — CHANGES_REQUESTED** [review]
Good job. Small change and we're good to go.

File: backend/src/main/java/ca/bc/gov/backendstartapi/enums/SeedlotStatus.java [hunk 2]
```diff
@@ -0,0 +1,26 @@
+package ca.bc.gov.backendstartapi.enums;
+
+import io.swagger.v3.oas.annotations.media.Schema;
+
+/** Possible status for Seed lots. */
+@Schema(description = "The possible status for a seed lot.")
+public enum SeedlotStatus implements DescribedEnum {
```
Comment (RMCampos) [inline backend/src/main/java/ca/bc/gov/backendstartapi/enums/SeedlotStatus.java:7]: I think we should keep the consistence, calling this enumeration `SeedlotStatusEnum`.

**Review by RMCampos — CHANGES_REQUESTED** [review]

File: backend/src/main/java/ca/bc/gov/backendstartapi/enums/SeedlotStatus.java (hunk 2 above)
Comment (annibalsilva) [reply backend/src/main/java/ca/bc/gov/backendstartapi/enums/SeedlotStatus.java]: Renamed.

**Pull request merged**